lib/ruby_core/core.rb

```diff
@@ -39,6 +39,7 @@ module RubyIntegration
         # TODO: fix int % float
         implement Fixnum, :%, PROMOTE_TO_FIXNUM_FLOAT, 1, "(#1 % #2)"
         implement Fixnum, :&, ASSERT_AND_RETURN_FIXNUM, 1, "(#1 & #2)"
+        implement Fixnum, :|, ASSERT_AND_RETURN_FIXNUM, 1, "(#1 | #2)"
         implement Fixnum, :*, PROMOTE_TO_FIXNUM_FLOAT, 1, "(#1 * #2)"
```

Comment:
I think we need parentheses around all variables. Otherwise expressions will not be evaluated in the correct order.